### DISCUSSION TITLE
patterns/proxy/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#56] [Announcements]
# patterns/proxy/

Also known as Surrogate Intent Provide a surrogate or placeholder for another object to control access to it. Explanation Real-world example Imagine a tower where the local wizards go to study their spells. The ivory tower can only be accessed through a proxy which ensures that only the first three wizards can enter. Here the proxy represents the functionality of the tower and adds access control to it.

https://java-design-patterns.com/patterns/proxy/

**Comment by moutainMan** [comment]
the best website of studying Java Design Patterns

**Comment by liuqTong** [comment]
this website's just a static proxy pattern.
Where can i find the dynamic proxy pattern (inplemented by java.lang.reflect.Proxy maybe fine) ? 
:)

**Reply by iluwatar** [reply]
Added a new issue https://github.com/iluwatar/java-design-patterns/issues/2542 Thanks for pointing this out @liuqTong 👍🏻
